modules/javafx.graphics/src/main/native-glass/gtk/glass_window.cpp

```diff
@@ -1010,14 +1010,10 @@ void WindowContextTop::process_configure(GdkEventConfigure* event) {
     }
 
     int x, y;
-
+    gdk_window_get_origin(gdk_window, &x, &y);
     if (frame_type == TITLED) {
-        GdkRectangle rect;
-        gdk_window_get_frame_extents(gdk_window, &rect);
```

Comment:
Looking at potential regression: in case of a TITLED frame, before this patch you would use `get_frame_extents` and now you will use `get_origin` in all cases. Are there scenario's where this might lead to regression, or is there a way to test this?

Reply:
`geometry.x` and `y` should point to the top left corner of the window including decorations (on JavaFX logic), then the view position will have the decoration size as x and y. `gdk_window_get_frame_extents` returns the top left position with frame extents (decorations) - all other Gdk/Gtk functions will return position without decorations (because on X11 they are drawn on client side). Somehow `gdk_window_get_frame_extents` is broken on i3-wm, but we already have the frame extents values. I just replaced it by getting normal origins (does not include decorations) and if the frame is TITLED, subtract the frame extents to have the window position accounting decorations.

To test this I:

- [ ] Run the bug example with UNDECORATED window on i3 and mutter (the default gnome wm);
- [ ] Run the bug example with DECORATED window on i3 and mutter (the default gnome wm);